File: packages/sdkwork-utils-java/src/main/java/com/sdkwork/utils/PathUtils.java
package com.sdkwork.utils;

import java.util.Arrays;
import java.util.stream.Collectors;

public final class PathUtils {
    private PathUtils() {
    }

    public static String joinPath(String... segments) {
        return Arrays.stream(segments)
                .map(segment -> segment.replaceAll("^/+|/+$", ""))
                .filter(segment -> !segment.isEmpty())
                .collect(Collectors.joining("/"));
    }

    public static String normalizePath(String value) {
        String joined = Arrays.stream(value.split("/"))
                .filter(part -> !part.isEmpty())
                .collect(Collectors.joining("/"));
        return value.startsWith("/") ? "/" + joined : joined;
    }
}

File: packages/sdkwork-utils-kotlin/src/main/kotlin/com/sdkwork/utils/PathUtils.kt
package com.sdkwork.utils

object PathUtils {
    fun joinPath(vararg segments: String): String {
        return segments
            .map { segment -> segment.replace(Regex("^/+|/+$"), "") }
            .filter { it.isNotEmpty() }
            .joinToString("/")
    }

    fun normalizePath(value: String): String {
        val joined = value.split("/")
            .filter { it.isNotEmpty() }
            .joinToString("/")
        return if (value.startsWith("/")) "/$joined" else joined
    }
}
